File: backend/src/main/java/pl/dudios/shop/common/model/Product.java
package pl.dudios.shop.common.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity
@Table(name = "products")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private Long categoryId;
    private String description;
    private String fullDescription;
    // Wersja angielska (opcjonalna) — null/pusty = frontend pokazuje polską.
    private String nameEn;
    private String descriptionEn;
    private String fullDescriptionEn;
    private BigDecimal price;
    private String currency;
    private String image;
    private String slug;
    @OneToMany
    @JoinColumn(name = "productId")
    private List<Review> reviews;

}

File: frontend/src/app/models/product.ts
export interface ProductDto {
  id: number;
  name: string;
  description: string;
  nameEn: string | null;
  descriptionEn: string | null;
  price: number;
  currency: string;
  image: string | null;
  slug: string;
}

export interface Review {
  id: number;
  productId: number;
  userId: number | null;
  authorName: string;
  content: string;
}

export interface Product {
  id: number;
  name: string;
  categoryId: number;
  description: string;
  fullDescription: string | null;
  nameEn: string | null;
  descriptionEn: string | null;
  fullDescriptionEn: string | null;
  price: number;
  currency: string;
  image: string | null;
  slug: string;
  reviews: Review[];
}

export interface ReviewRequest {
  authorName: string;
  content: string;
  productId: number;
}
